lib/spack/spack/repo.py

```diff
@@ -1516,7 +1517,19 @@ def __init__(self, name, repo=None):
                 long_msg = "Did you mean to specify a filename with './{0}'?"
                 long_msg = long_msg.format(name)
             else:
-                long_msg = "You may need to run 'spack clean -m'."
+                long_msg = "You may need to run 'spack clean -m'"
+                long_msg += " or use 'spack create' to create a new package."
```

Comment:
Can we remove this? At one time it was very common for issues with cached packages that were renamed. Nowadays this is less common and most of the time it was simply a typo.

Reply:
Done, I've kept the `spack create` suggestion for now, should we also remove that?

Reply:
I'm fine with keeping it. `spack clean` just felt like we were documenting a bug.